src/Features/Core/SolutionCrawler/WorkCoordinator.GlobalOperationAwareIdleProcessor.cs

```diff
@@ -113,25 +113,34 @@ private void OnGlobalOperationStopped(object sender, GlobalOperationEventArgs e)
 
                     protected async Task WaitForHigherPriorityOperationsAsync()
                     {
-                        do
+                        using (Logger.LogBlock(FunctionId.WorkCoordinator_WaitForHigherPriorityOperationsAsync, this.CancellationToken))
                         {
-                            // we wait for global operation and higher queue operation if there is anything going on
-                            if (!this.GlobalOperationTask.IsCompleted || !this.HigherQueueOperationTask.IsCompleted)
+                            do
                             {
-                                await Task.WhenAll(this.GlobalOperationTask, this.HigherQueueOperationTask).ConfigureAwait(false);
+                                // Host is shutting down
+                                if (this.CancellationToken.IsCancellationRequested)
+                                {
+                                    return;
+                                }
+
+                                // we wait for global operation and higher queue operation if there is anything going on
+                                if (!this.GlobalOperationTask.IsCompleted || !this.HigherQueueOperationTask.IsCompleted)
+                                {
+                                    await Task.WhenAll(this.GlobalOperationTask, this.HigherQueueOperationTask).ConfigureAwait(false);
```

Comment:
WhenAll doesn't accept cancellation. but the Task given to WhenAll already have the cancellation in them.